checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,371 @@
+
+    /**
+     * A record reader converting input to record-oriented view.
+     */
+    static class CheckerRecordReader
+        extends RecordReader<IntWritable, NullWritable> {
+      int mIntegerVal;
+      IntWritable mKey = null;
+
+      public CheckerRecordReader() {
+      }
+
+      @Override
+      public void initialize(InputSplit split, TaskAttemptContext context) {
+        mIntegerVal = ((CheckerInputSplit) split).mValue;
+      }
+
+      @Override
+      public void close() {
+        // NOTHING
+      }
+
+      @Override
+      public IntWritable getCurrentKey() {
+        return mKey;
+      }
+
+      @Override
+      public NullWritable getCurrentValue() {
+        return NullWritable.get();
+      }
+
+      @Override
+      public float getProgress() {
+        if (sCreateDone) {
+          return 1.0f;
+        } else {
+          return 0.0f;
+        }
+      }
+
+      @Override
+      public boolean nextKeyValue() {
+        if (mKey == null) {
+          mKey = new IntWritable(mIntegerVal);
+        }
+        if (sCreateDone) {
+          return false;
+        } else {
+          sCreateDone = true;
+          return true;
+        }
+      }
+    }
+
+    @Override
+    public RecordReader<IntWritable, NullWritable> createRecordReader(InputSplit split,
+        TaskAttemptContext context) {
+      return new CheckerRecordReader();
+    }
+
+    /**
+     * Creates the desired number of splits.
+     */
+    @Override
+    public List<InputSplit> getSplits(JobContext job) {
+      int numSplits = job.getConfiguration().getInt(MRJobConfig.NUM_MAPS, 1);
+      List<InputSplit> splits = new ArrayList<>();
+      for (int split = 0; split < numSplits; split++) {
+        splits.add(new CheckerInputSplit(1));
+      }
+      return splits;
+    }
+  }
+
+  /**
+   * In each mapper node, we will check whether this node can recognize
+   * Alluxio classes and filesystem.
+   */
+  protected static class CheckStatusMapper extends Mapper<IntWritable, NullWritable, Text, Text> {
+    /**
+     * Records the Status and IP address of each mapper task node.
+     */
+    @Override
+    protected void map(IntWritable key, NullWritable value, Context context)
+        throws IOException, InterruptedException {
+      context.write(new Text(CheckerUtils.performIntegrationChecks().toString()),
+          new Text(CheckerUtils.getLocalAddress()));
+    }
+  }
+
+  /**
+   * In each reducer node, we will combine the IP addresses that have the same Status.
+   */
+  protected static class MergeStatusReducer extends Reducer<Text, Text, Text, Text> {
+    /**
+     * Merges the IP addresses of same Status.
+     */
+    @Override
+    protected void reduce(Text key, Iterable<Text> values, Context context)
+        throws IOException, InterruptedException {
+      Set<String> addressSet = new HashSet<>();
+      for (Text val : values) {
+        String curAddress = val.toString();
+        if (!addressSet.contains(curAddress)) {
```

Comment:
as `addressSet` is a set, we don't need to check the membership again right? 
i.e. `if (!addressSet.contains(curAddress))` is not helping